### PR TITLE
RTL: some small fixes for fixed-wing/VTOL

### PR DESCRIPTION
**Pull request by sfuhrer**
## Describe problem solved by this pull request
- loiter radius and direction of the loiter down in a RTL during a mission landing could differate from loiter radius and direction before mission landing was engaged. 
image
- loiter (orbits) with negative loiter directions (CCW) were not correctly displayed on the GS
- Orbit status of loiter before landing was not shown until on loiter (as WP wasn't of type Loiter) and vehicle flew through center of loiter down instead of joining it smoothly:
image

## Describe your solution
- if RTL is of type RTL_DESTINATION_MISSION_LANDING, then set loiter radius and direction of Descend loiter (where it will descend to RTL_DESCEND_ALT and wait there for RTL_DELAY seconds) to what was set in the mission.
- uint8_t -->int8_t
- for FW flight: already set WP type to LOITER_TO_ALT for state RTL_STATE_RETURN.
image

## Test data / coverage
SITL (standard VTOL and Plane)

### CODE REVIEW

File: src/modules/navigator/rtl.cpp [hunk 1]
```diff
@@ -423,9 +434,8 @@ void RTL::set_rtl_item()
 				_mission_item.yaw = _destination.yaw;
 			}
 
-			if (_navigator->get_vstatus()->is_vtol
-			    || _navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING) {
-				_mission_item.loiter_radius = _rtl_loiter_rad;
+			if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING) {
+				_mission_item.loiter_radius = landing_loiter_radius;
```
Comment (tstastny) [inline src/modules/navigator/rtl.cpp:438]: should we explicitly set the loiter radius here even if it is not a fixed-wing? it is not set before the switch case, and in this case the waypoint type is a loiter. is it already defaulted somewhere else?

File: src/modules/navigator/rtl.cpp [hunk 2]
```diff
@@ -471,7 +481,10 @@ void RTL::set_rtl_item()
 				_mission_item.yaw = _destination.yaw;
 			}
 
-			_mission_item.loiter_radius = _navigator->get_loiter_radius();
+			if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING) {
+				_mission_item.loiter_radius = landing_loiter_radius;
+			}
```
Comment (tstastny) [inline src/modules/navigator/rtl.cpp:486]: same question

**Review by tstastny — CHANGES_REQUESTED** [review]
Small question on the loiter radius population - otherwise good catch! (and thanks for cleaning up at least one internal state :) )

File: src/modules/navigator/rtl.cpp (hunk 2 above)
Comment (sfuhrer) [reply src/modules/navigator/rtl.cpp]: https://github.com/PX4/PX4-Autopilot/pull/20214/commits/b9ddb4510936d5da3d10b662b9f130ec63b0a7a5

File: src/modules/navigator/rtl.cpp (hunk 1 above)
Comment (sfuhrer) [reply src/modules/navigator/rtl.cpp]: You're right, let's just always fill all the fields, independent on vehicle type.
https://github.com/PX4/PX4-Autopilot/pull/20214/commits/b9ddb4510936d5da3d10b662b9f130ec63b0a7a5

**Review by tstastny — APPROVED** [review]

**Review by dagar — APPROVED** [review]

**Pull request merged**